algoliasearch-core/src/test/java/com/algolia/search/integration/client/ApiKeysTest.java

```diff
@@ -62,7 +63,17 @@ void testApiKeys() {
     DeleteApiKeyResponse deleteApiKey = searchClient.deleteApiKeyAsync(apiKey).join();
     deleteApiKey.waitTask();
 
-    RestoreApiKeyResponse restoreApiKey = searchClient.restoreApiKeyAsync(apiKey).join();
+    RestoreApiKeyResponse restoreApiKey;
+    while (true) {
+      try {
+        restoreApiKey = searchClient.restoreApiKeyAsync(apiKey).join();
```

Comment:
Not if it is important in this case, but maybe a retry count logic here to avoid retrying indefinitely?

Reply:
Since most of our `.waitTask()` methods are blocking without time limitation, I thought it wouldn't matter here for the test. But you're right, this is good practice anyway. I'll add a `retry()` test helper and use it here 👍